libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -67,7 +67,27 @@ public virtual async Task LoadAsync(ITurnContext turnContext, bool force = false
             {
                 var items = await _storage.ReadAsync(new[] { storageKey }, cancellationToken).ConfigureAwait(false);
                 items.TryGetValue(storageKey, out object val);
-                turnContext.TurnState[_contextServiceKey] = new CachedBotState((IDictionary<string, object>)val);
+
+                if (val is IDictionary<string, object> asDictionary)
+                {
+                    turnContext.TurnState[_contextServiceKey] = new CachedBotState(asDictionary);
+                }
+                else if (val is JObject asJobject)
+                {
+                    // If types are not used by storage serialization, and Newtonsoft is the serializer
+                    // the item found will be a JObject.
+                    turnContext.TurnState[_contextServiceKey] = new CachedBotState(asJobject.ToObject<IDictionary<string, object>>());
+                }
+                else if (val is null)
```

Comment:
This seems like an odd test, it _works_ buts that's more of a quirk of the language. I think what you meant was:
  if (vall == null)

Reply:
changed to '=='